### PR TITLE
Ensure OpenAI GPT position_ids is correctly initialized and registered at init.

### PR DESCRIPTION
**Pull request by mfuntowicz**
This will make it compatible with TorchScript export and avoid hardcoded `position_ids` tensor's device in the generated graph.

Solve the following issue: #5664 	

Signed-off-by: Morgan Funtowicz <morgan@huggingface.co>

### CODE REVIEW

**Comment by mfuntowicz** [comment]
Fixes for the failing tests are not detected by Github while being present on the branch ... Lets see if it comes back to life in a while ...

File: src/transformers/modeling_openai.py
Comment (sshleifer) [inline src/transformers/modeling_openai.py:447]: Does this need to happen in other places? I'm confused why only openaigpt is failing.

**Review by patrickvonplaten — APPROVED** [review]
Nice!

File: src/transformers/modeling_openai.py
Comment (patrickvonplaten) [reply src/transformers/modeling_openai.py:447]: Yeah, I'd also be interested if this only happens for openai or other models as well...

Comment (mfuntowicz) [reply src/transformers/modeling_openai.py:447]: That's possible for some other models the same kind of issue. We need to discuss what kind of level of support we want to give to TorchScript, in such case we might need to deploy a broader unittests set as we do for ONNX

**Comment by mfuntowicz** [comment]
CI failure has been fixed on master.

File: src/transformers/modeling_openai.py
Comment (mfuntowicz) [reply src/transformers/modeling_openai.py:447]: @sshleifer I think we're not actively testing export to TorchScript so we're not monitoring issue(s) on this aspect.

Comment (vdantu) [reply src/transformers/modeling_openai.py:447]: Should we make changes to these files as well?

[modeling_bert.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_bert.py#L191-L192)
[modeling_mobilebert.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_mobilebert.py#L189-L191)
[modeling_xlm.py](https://github.com/huggingface/transformers/blob/63761614eb7a5f96e089b77d5469ea31a7177e16/src/transformers/modeling_xlm.py#L517-L519)

Disclaimer: I did a cursory code match, so please ignore if this doesn't apply.

Comment (mfuntowicz) [reply src/transformers/modeling_openai.py:447]: I can address all of them on this PR @patrickvonplaten @sshleifer

**Comment by mfuntowicz** [comment]
@sshleifer I've updated all the models with `position_ids` created in the forward pass as it seems to be the right design to allow people to export the bunch of models impacted. 

Also, it avoids allocating tensors at every forward call, so might reduce the pressure on PyTorch memory manager.

**Comment by sshleifer** [comment]
Do we get the "Uninitialized parameters" warning because the position_ids are not in the `pytorch_model.bin` on S3?

**Comment by mfuntowicz** [comment]
I'll check the above, didn't catch at first 👌

**Comment by vdantu** [comment]
Could this be merged? Would love to test these changes after merging :)

**Comment by mfuntowicz** [comment]
@vdantu Now merged, sorry for the delay, many people are off these days, it slows down a little bit the merging process 😃. 

Let us know if you have any follow up issue(s) / question(s).

**Comment by vdantu** [comment]
Thanks for the push. Would we have to wait for a new release of `transformers` package to get these changes?

**Comment by mfuntowicz** [comment]
@vdantu: I'll let @LysandreJik handle this one

File: src/transformers/modeling_openai.py
Comment (vadimkantorov) [reply src/transformers/modeling_openai.py:447]: we also hit the same problem in gpt_2: https://github.com/huggingface/transformers/issues/5664#issuecomment-675511529

Comment (vadimkantorov) [reply src/transformers/modeling_openai.py:447]: if this indeed fixed the tracing issues, then the fix should be same for modeling_gpt2.py

**Pull request merged**
